src/Service/S3/src/S3Client.php

```diff
@@ -973,7 +973,8 @@ protected function getAwsErrorFactory(): AwsErrorFactoryInterface
     protected function getEndpoint(string $uri, array $query, ?string $region): string
     {
         $uriParts = explode('/', $uri, 3);
-        $bucket = $uriParts[1] ?? '';
+        $bucket = preg_replace('~\?.*~', '', $uriParts[1] ?? '');
+        $uriWithOutBucket = preg_replace('~^[^?]+~', '', $uriParts[1] ?? '') . ($uriParts[2] ?? '');
```

Comment:
no need to use regexp
```suggestion
        $uriWithOutBucket = substr($uriParts[1] ?? '', strlen($bucket)) . ($uriParts[2] ?? '');
```